src/core/server/elasticsearch/client/configure_client.ts

```diff
@@ -39,10 +38,8 @@ const addLogging = (client: Client, logger: Logger, logQueries: boolean) => {
   client.on('response', (error, event) => {
     if (error) {
       const errorMessage =
-        // error details for response errors provided by elasticsearch
-        isResponseError(error)
-          ? `[${event.body.error.type}]: ${event.body.error.reason}`
-          : `[${error.name}]: ${error.message}`;
+        // error details for response errors provided by elasticsearch, defaults to error name/message
+        `[${event.body?.error?.type ?? error.name}]: ${event.body?.error?.reason ?? error.message}`;
```

Comment:
looks`?`scary`?` 😄

Reply:
TBH I started with

```ts
isResponseError(error) && event.body?.error
          ? `[${event.body.error.type}]: ${event.body.error.reason}`
          : `[${error.name}]: ${error.message}`;
```

but the current version felt slightly better. Can switch back if we want.

Reply:
no, it's fine